server/aesdsocket.c

```diff
@@ -0,0 +1,355 @@
+  if(file_des == -1) //file_ptr
+  {
+    syslog(LOG_ERR,"file creation,opening failure\n");
+    perror("file creation");
+    exit(1);
+  }
+  
+  /* local declarations */
+  int current_data_pos = 0;
+  int read_status = 0;
+  int no_of_bytes_rcvd = 0;
+  
+  char* writer_file_buffer_ptr = NULL;
+  int write_buffer_size = BUFFER_CAPACITY;
+  writer_file_buffer_ptr = (char *)malloc(sizeof(char)*BUFFER_CAPACITY);
+  memset(writer_file_buffer_ptr,'\0',BUFFER_CAPACITY);
+  if(writer_file_buffer_ptr == NULL)
+  {
+    perror("writer_file_buffer");
+  }
+  
+  char* read_file_buffer_ptr = NULL;
+  bool run_status = true;
+  
+  while(run_status)
+  {    
+    /* Accept */
+    int client_accept_fd = 0;
+    socklen_t server_address_len = 0;
+    server_address_len = sizeof(server_address);
+    char s[INET6_ADDRSTRLEN];
+    memset(&s[0],'\0',sizeof(s));
+    
+    client_accept_fd = accept(server_socket_fd, (struct sockaddr *)&server_address,&server_address_len);
+    /* Accepted connection from client */
+    inet_ntop(AF_INET,get_in_addr((struct sockaddr *)&server_address),s,sizeof(s));
+    syslog(LOG_DEBUG, "Accepted connection from %s", s);
+    if(client_accept_fd == -1)
+    {
+      perror("server_accept_fd");
+      continue;
+    }
+    
+    /* sig status */
+    int sig_status = 0;
+    sig_status = sigprocmask(SIG_BLOCK, &x, NULL);
+    if(sig_status == -1)
+    {
+      perror("sig_status - 1");
+    }
+    
+    /* read */
+    int curr_location = 0;
+    while((no_of_bytes_rcvd = (recv(client_accept_fd, writer_file_buffer_ptr + curr_location, BUFFER_CAPACITY, 0))))
+    {
+      if(no_of_bytes_rcvd + curr_location >= write_buffer_size)
+      {
+        write_buffer_size *= MULTIPLIER_FACTOR; 
+        //printf(" ");
+        syslog(LOG_DEBUG,"write_buffer_size = %d\n",write_buffer_size);
+        char* tmp_ptr = realloc(writer_file_buffer_ptr, write_buffer_size);
+        
+        if(tmp_ptr == NULL)
+        {
+          /* memory allocation failure. */
+          perror("realloc failure");
+          free(writer_file_buffer_ptr);
+          writer_file_buffer_ptr = NULL;
+          run_status = false;
+          break;
+        }
+         writer_file_buffer_ptr = tmp_ptr;
+      }
+      
+      curr_location = curr_location + no_of_bytes_rcvd;
+      
+      if(strstr(writer_file_buffer_ptr,"\n") != NULL)
+      {
+        //Take a break when "\n" is detected (i..e termination conditon) or 
+        //when the no of bytes received are zero
+        break;
+      }
+    }
+    
+    ret_status = write(file_des,writer_file_buffer_ptr,curr_location);
+    if(ret_status == -1)
+    {
+      syslog(LOG_ERR,"file writing failure\n");
+      perror("file writing");
+    }
+    
+    /* lseek required to bring to start position*/
+    current_data_pos = lseek(file_des,0,SEEK_CUR);
+    syslog(LOG_DEBUG,"position is %d\n",current_data_pos);
+    
+    read_file_buffer_ptr = (char *) malloc(sizeof(char)*current_data_pos);
```

Comment:
@Rishab-Shah what assumption is made about the size of the RAM heap relative to the total size of all packets sent?